urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/tools/load/view/LoadPanel.java

```diff
@@ -0,0 +1,1359 @@
+				contextShift = (patch.start2 + patch.length2) - (patch.start1 + patch.length1);
+			} catch (BadLocationException ble) {
+				ble.printStackTrace();
+			}
+		}
+
+		// Highlight the revised text
+		for (Patch patch : patches) {
+			// Highlight the original text
+			DefaultHighlighter.DefaultHighlightPainter highlightPainter = new DefaultHighlighter.DefaultHighlightPainter(Color.YELLOW);
+			try {
+				textAreaNew.getHighlighter().addHighlight(patch.start2 + 4, patch.start2 + patch.length2 - 4, highlightPainter);
+			} catch (BadLocationException ble) {
+				ble.printStackTrace();
+			}
+		}
+	}
+
+	private void openPreviewPanel() throws PtoolsErrorException {
+		if (controller.isKBModified(controller.getSelectedOrganism())) {
+			int n = JOptionPane.showConfirmDialog(
+					DefaultController.mainJFrame,
+			    "The selected database is already in a modified state. \n" +
+				"Recommend saving or undoing changes to this database \n" +
+			    "before making additional changes. \n\n" +
+				"Do you wish to continue anyway?",
+			    "Database in Modified State",
+			    JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
+			if (n == 1) {
+				return;
+			}
+		}
+		textAreaOld.setText("");
+		textAreaNew.setText("");
+		
+		DefaultTableModel model = new DefaultTableModel();
+		if (cmbAdaptor.getSelectedIndex() == 3) {
+			model = getDataFromFile();
+		} else model = getDataFromMatches();
+	    
+		if (model.getRowCount() == 0) {
+			JOptionPane.showMessageDialog(this, "No data could be matched to frames in this database for the selected frame type. Unable to proceed with import", "No Data to Import!", JOptionPane.WARNING_MESSAGE);
+			return;
+		}
+	    
+	    batchEdits = new BatchUpdate(selectedAdaptor.tableToFrameUpdates(model), this);
+		batchEdits.addPropertyChangeListener(controller);
+		
+		batchEdits.downloadFrames(controller.getConnection());
+		allFramesWithImports = batchEdits.getFrameIDsModel();
+		listFrames.setModel(allFramesWithImports);
+	}
+	
+	public DefaultTableModel getDataFromFile() {
+		DefaultTableModel model = new DefaultTableModel();
+		DefaultTableModel dtmFile = (DefaultTableModel) tableSpreadSheet.getModel();
+		int nRow = dtmFile.getRowCount();
+	    int nCol = dtmFile.getColumnCount();
+	    
+	    Object[] tableHeaders = new Object[nCol];
+	    for (int i = 1; i < nCol; i++) {
+			if (i < 1) tableHeaders[i] = tableSpreadSheet.getColumnModel().getColumn(i-1).getHeaderValue();
+	    }
+	    
+	    if (nCol != 0) { // If no data, then don't try to parse it out
+		    Object[][] tableData = new Object[nRow][nCol];
+			
+		    for (int i = 0 ; i < nRow; i++) {
+		        for (int j = 0 ; j < nCol ; j++) {
+		        	tableData[i][j] = dtmFile.getValueAt(i,j);
+		        }
+		    }
+		    
+		    model = new DefaultTableModel(tableData, tableHeaders);
+	    }
+	    
+	    return model;
+	}
+	
+	public DefaultTableModel getDataFromMatches() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.